File: src/cpp/1747.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <cmath>
using namespace std;

int main() {
	ios_base::sync_with_stdio(0); cin.tie(0);
	int i;
	cin >> i;

	while (true)
	{
		if (i <= 2) {
			cout << "2";
			break;
		}
		bool prime = true;
		for (int j = 2; j <= ceil(sqrt(i)); j++)
		{
			if (i % j == 0)
			{
				prime = false;
				break;
			}
		}
		if (prime)
		{
			string d = to_string(i);
			if (d.front() == d.back())
			{
				string copy_d = d;
				reverse(copy_d.begin(), copy_d.end());
				if (d == copy_d)
				{
					cout << d;
					break;
				}
			}
		}
		i++;
	}
	return 0;
}

File: src/python/1747.py
from sys import stdin, stdout
import math

i = int(stdin.readline())
while True:
    if i == 1:
        continue
    if i == 2:
        stdout.write('2')
        break
    prime = True
    for j in range(2, math.ceil(math.sqrt(i)) + 1):
        if i % j == 0:
            prime = False
            break
    if prime:
        if str(i)[0] == str(i)[-1]:
            if str(i) == str(i)[::-1]:
                stdout.write(f"{i}")
                break
    i += 1